db/trie/mptrie/node.go

```diff
@@ -17,20 +17,28 @@ var ErrNoData = errors.New("no data in hash node")
 type (
 	keyType []byte
 
+	client interface {
+		asyncMode() bool
+		hash([]byte) []byte
+		loadNode([]byte) (node, error)
+		deleteNode([]byte) error
+		putNode([]byte, []byte) error
+	}
+
 	node interface {
-		Search(keyType, uint8) (node, error)
-		Delete(keyType, uint8) (node, error)
-		Upsert(keyType, uint8, []byte) (node, error)
-		Hash() ([]byte, error)
-		Flush() error
+		Search(client, keyType, uint8) (node, error)
+		Delete(client, keyType, uint8) (node, error)
+		Upsert(client, keyType, uint8, []byte) (node, error)
+		Hash(client) ([]byte, error)
+		Flush(client) error
```

Comment:
is it possible not add `client` to the func's input, if the node struct (leaf, extension, branch) contains the `client` interface?
feel it's a bit heavy/clumsy with this extra input